### PR TITLE
AZ-383 fast accounts initialization for flooder

### PR DESCRIPTION
**Pull request by fixxxedpoint**
- allows to skip accounts initialization
- allows to provide first index for the set of integer derived accounts, i.e. accounts derived from range x.. instead of the default 0.. range
- general refactor

### CODE REVIEW

**Review by fbielejec — APPROVED** [review]

**Review by kostekIV — APPROVED** [review]
lgtm

File: flooder/src/config.rs [hunk 1]
```diff
@@ -23,6 +23,18 @@ pub struct Config {
...
+    #[clap(long)]
+    pub skip_initialization: bool,
+
+    /// beginning of the integer range used to derive accounts
+    #[clap(long, default_value = "0")]
+    pub first_account_in_range: u64,
+
+    /// number of threads spawn during the flooding process
+    #[clap(long)]
+    pub threads: Option<u64>,
```
[the remark this answers is not in the document]
Comment (fixxxedpoint) [reply flooder/src/config.rs:33]: I hate using platform dependent types in external apis.

File: flooder/src/config.rs [hunk 2]
```diff
@@ -23,6 +23,18 @@ pub struct Config {
     /// secret seed of the account keypair passed on stdin
     #[clap(long, conflicts_with_all = &["phrase"])]
     pub seed: Option<String>,
+
+    /// allows to skip accounts initialization process and just attempt to download their nonces
+    #[clap(long)]
+    pub skip_initialization: bool,
```
[the remark this answers is not in the document]
Comment (fixxxedpoint) [reply flooder/src/config.rs:25]: I don't know what is more intuitive. Flag is more verbose and is also described using the `--help` parameter.

File: flooder/src/main.rs [hunk 3]
```diff
@@ -28,43 +30,82 @@ fn main() -> Result<(), anyhow::Error> {
...
+    }
+
+    let nonces = accounts
+        .iter()
+        .map(|account| get_nonce(&connection, &AccountId::from(account.public())))
+        .collect();
 
     debug!("all accounts have received funds");
 
+    let reciever = accounts
```
[the remark this answers is not in the document]
Comment (fixxxedpoint) [reply flooder/src/main.rs]: changed

File: flooder/src/config.rs [hunk 4]
```diff
@@ -23,6 +23,18 @@ pub struct Config {
     /// secret seed of the account keypair passed on stdin
     #[clap(long, conflicts_with_all = &["phrase"])]
     pub seed: Option<String>,
+
```
[the remark this answers is not in the document]
Comment (fixxxedpoint) [reply flooder/src/config.rs:22]: removed

**Review by mike1729 — APPROVED** [review]

**Pull request merged**